### PR TITLE
Version watcher

### PR DESCRIPTION
**Pull request by qasimkhan**
Smartstack related changes: https://github.com/dubizzle/smartstack/pull/5/

### CODE REVIEW

File: lib/synapse/service_watcher/version.rb [hunk 1]
```diff
@@ -0,0 +1,90 @@
+require_relative "zookeeper"
+
+require 'zk'
+
+module Synapse
+  class VersionWatcher < ZookeeperWatcher
```
Comment (dcrasto) [inline lib/synapse/service_watcher/version.rb:6]: Good OO inheriting from ZookeeperWatcher

File: lib/synapse/service_watcher/version.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      @version_callback ||= Proc.new do |event|
+        # Set new watcher
+        version_watch
+        # Rediscover
+        version_discover
+        # restart synapse
+        if @restart_synapse
+          @restart_synapse = false
+          log.info("restarting synapse")
+          res = `#{@discovery['reload_command']}`
```
Comment (dcrasto) [inline lib/synapse/service_watcher/version.rb:82]: works just like bash huh. Pretty cool.

**Comment by dcrasto** [comment]
looks got to go

File: lib/synapse/service_watcher/version.rb [hunk 3]
```diff
@@ -0,0 +1,90 @@
+require_relative "zookeeper"
+
+require 'zk'
+
+module Synapse
+  class VersionWatcher < ZookeeperWatcher
+    def start
+      zk_hosts = @discovery['hosts'].shuffle.join(',')
+
+      log.info "synapse: starting ZK watcher #{@name} @ hosts: #{zk_hosts},
```
Comment (Dareen) [inline lib/synapse/service_watcher/version.rb]: "starting version watcher" instead of "ZK watcher"

**Comment by Dareen** [comment]
@dcrasto would you please merge this PR

**Pull request merged**